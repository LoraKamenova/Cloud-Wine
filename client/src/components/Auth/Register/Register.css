.register-form-wrapper > .form-header-wrapper > .form-header-heading-container > .form-header-heading {
    bottom: -25px;
    left: 15px
}

.form-wrapper.register-form-wrapper {
    height: 95%;
    margin: auto;
}
