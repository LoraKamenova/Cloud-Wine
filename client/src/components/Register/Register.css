.register-container {
    width: 60%;
    height: 92%;
    margin: auto;
}
