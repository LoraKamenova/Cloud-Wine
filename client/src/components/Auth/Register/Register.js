import './Register.css'
import FormHeader from "../../Shared/FormHeader/FormHeader";

import {Component} from 'react';

import authenticate from "../../../utils/authenticate";
import UserContext from "../../../Context";

class Register extends Component {

    constructor(props) {
        super(props)

        this.state = {
            username: "",
            password: "",
            avatarUrl: "",
            errorMessage: "",
        }
    }

    static contextType = UserContext;

    onChange = (event, type) => {
        const newState = {}
        newState[type] = event.target.value

        this.setState(newState)
    }

    handleSubmit = async (event) => {
        event.preventDefault()
        const {
            username,
            password,
            avatarUrl
        } = this.state

        await authenticate('http://localhost:5000/api/user/register', {
                username,
                password,
                avatarUrl
            }, (user) => {
                this.context.logIn(user)
                this.props.history.push('/home')
            }, (e) => { this.setState({errorMessage: "Потребителското име е заето"})}
        )
    }

    render() {
        const {
            username,
            avatarUrl,
            password,
        } = this.state

        let errorSpan = '';
        if(this.state.errorMessage) {
            errorSpan = <span className="form-error-span">{this.state.errorMessage}</span>
        }

        return (
            <section className="form-wrapper register-form-wrapper">
                <FormHeader title="Регистрация"/>
                <form className="form-container" onSubmit={this.handleSubmit}>

                    <div className="form-field-wrapper">
                        <div className="form-field-heading">
                            <label htmlFor="username">Потребителско име:*</label>
                        </div>
                        <input className="form-field-input"
                               type="text"
                               id="username"
                               name="username"
                               required
                               value={username}
                               onChange={(e) => this.onChange(e, 'username')}/>
                    </div>

                    {/*<div className="form-field-wrapper">*/}
                    {/*    <div className="form-field-heading">*/}
                    {/*        <label htmlFor="rePassword">Повторете паролата:*</label>*/}
                    {/*    </div>*/}
                    {/*    <input className="form-field-input"*/}
                    {/*           type="password"*/}
                    {/*           id="rePassword"*/}
                    {/*           name="rePassword"*/}
                    {/*           required*/}
                    {/*           value={rePassword}*/}
                    {/*           onChange={(e) => this.onChange(e, 'rePassword')}/>*/}
                    {/*</div>*/}

                    <div className="form-field-wrapper">
                        <div className="form-field-heading">
                            <label htmlFor="avatarUrl">URL профилна снимка:</label>
                        </div>
                        <input className="form-field-input"
                               type="avatarUrl"
                               id="avatarUrl"
                               name="avatarUrl"
                               value={avatarUrl}
                               onChange={(e) => this.onChange(e, 'avatarUrl')}/>
                    </div>

                    <div className="form-field-wrapper">
                        <div className="form-field-heading">
                            <label htmlFor="password">Парола:*</label>
                        </div>
                        <input className="form-field-input"
                               type="password"
                               id="password"
                               name="password"
                               required
                               value={password}
                               onChange={(e) => this.onChange(e, 'password')}/>
                    </div>

                    <span className="form-info-span">Полетата, отбелязани със (*) са задъжлителни</span>

                    <div className="form-error-div">
                        {errorSpan}
                    </div>

                    <div className="form-button-wrapper">
                        <input className="form-button" type="submit" value="ИЗПРАТИ"/>
                    </div>
                </form>
            </section>
        )
    }
}

export default Register;
