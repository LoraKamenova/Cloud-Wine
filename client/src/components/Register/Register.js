import './Register.css'
import FooterBottom from "../FooterBottom/FooterBottom";

const Register = () => {
    return (
        <main className="auth-pages register-page">

            <section className="auth-form-wrapper register-form-wrapper">

                <div className="auth-form-site-name-wrapper">
                    <div className="auth-form-site-name-text">
                        <h1 className="auth-form-site-name">Винoписи</h1>
                    </div>
                    <div className="auth-form-site-name-pic">
                        <img className="auth-form-pic"
                             src="https://res.cloudinary.com/dyhmxus4n/image/upload/v1615712293/React%20Project/grape_border_ogd8gq.png"/>
                    </div>
                </div>

                <form className="auth-form" action="" method="">
                    <div className="auth-form-heading-wrapper">
                        <h1 className="auth-form-heading">РЕГИСТРАЦИЯ</h1>
                    </div>

                    <div className="auth-form-field-wrapper">
                        <div className="auth-form-field-heading">
                            <label htmlFor="username">Потребителско име:</label>
                        </div>
                        <input className="auth-form-field-input" type="text" id="username" name="username"/>
                    </div>

                    <div className="auth-form-field-wrapper">
                        <div className="auth-form-field-heading">
                            <label htmlFor="email">Имейл адрес:</label>
                        </div>
                        <input className="auth-form-field-input" type="email" id="email" name="email"/>
                    </div>

                    <div className="auth-form-field-wrapper">
                        <div className="auth-form-field-heading">
                            <label htmlFor="password">Парола:</label>
                        </div>
                        <input className="auth-form-field-input" type="password" id="password" name="password"/>
                    </div>

                    <div className="auth-form-field-wrapper">
                        <div className="auth-form-field-heading">
                            <label htmlFor="re-password">Повторете паролата:</label>
                        </div>
                        <input className="auth-form-field-input" type="password" id="re-password" name="re-password"/>
                    </div>

                    <div className="auth-form-button-wrapper">
                        <button className="auth-form-button" type="submit">ИЗПРАТИ</button>
                    </div>
                </form>
            </section>
            <FooterBottom />
        </main>
    );
};

export default Register;
