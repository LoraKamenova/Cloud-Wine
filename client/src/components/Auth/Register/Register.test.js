import { render } from '@testing-library/react';
import Register from './Register';

describe('Register Component', () => {

    it("Register Component snapshot", () => {
        const { asFragment } = render(<Register />);
        expect(asFragment()).toMatchSnapshot();
    });

});
