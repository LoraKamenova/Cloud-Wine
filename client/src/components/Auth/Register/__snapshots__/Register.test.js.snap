// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Register Component Register Component snapshot 1`] = `
<DocumentFragment>
  <section
    class="form-wrapper register-form-wrapper"
  >
    <div
      class="form-header-wrapper"
    >
      <div
        class="form-header-heading-container"
      >
        <h1
          class="form-header-heading"
        >
          Регистрация
        </h1>
      </div>
      <div
        class="form-header-image-container"
      >
        <img
          alt="vine"
          class="form-header-image"
          src="https://res.cloudinary.com/dyhmxus4n/image/upload/v1615712293/React%20Project/grape_border_ogd8gq.png"
        />
      </div>
      <h1
        class="form-header-heading-hidden"
      >
        Регистрация
      </h1>
    </div>
    <form
      class="form-container"
    >
      <div
        class="form-field-wrapper"
      >
        <div
          class="form-field-heading"
        >
          <label
            for="username"
          >
            Потребителско име:*
          </label>
        </div>
        <input
          class="form-field-input"
          id="username"
          name="username"
          required=""
          type="text"
          value=""
        />
      </div>
      <div
        class="form-field-wrapper"
      >
        <div
          class="form-field-heading"
        >
          <label
            for="password"
          >
            Парола:*
          </label>
        </div>
        <input
          class="form-field-input"
          id="password"
          name="password"
          required=""
          type="password"
          value=""
        />
      </div>
      <div
        class="form-field-wrapper"
      >
        <div
          class="form-field-heading"
        >
          <label
            for="rePassword"
          >
            Повторете паролата:*
          </label>
        </div>
        <input
          class="form-field-input"
          id="rePassword"
          name="rePassword"
          required=""
          type="password"
          value=""
        />
      </div>
      <div
        class="form-field-wrapper"
      >
        <div
          class="form-field-heading"
        >
          <label
            for="avatarUrl"
          >
            URL профилна снимка:
          </label>
        </div>
        <input
          class="form-field-input"
          id="avatarUrl"
          name="avatarUrl"
          type="avatarUrl"
          value=""
        />
      </div>
      <span
        class="form-info-span"
      >
        Полетата, отбелязани със (*) са задъжлителни
      </span>
      <div
        class="form-error-div"
      />
      <div
        class="form-button-wrapper"
      >
        <input
          class="form-button auth-form-button"
          type="submit"
          value="ИЗПРАТИ"
        />
      </div>
    </form>
  </section>
</DocumentFragment>
`;
